client/src/services/link.js

```diff
@@ -49,181 +44,134 @@ export function LinkProvider(props) {
+      await postLinkEvent({
+        userId,
+        link_session_id,
+        type: 'success',
+      });
       if (isUpdate) {
-        hasRequested.current.byItem[itemId] = true;
+        await setItemState(itemId, 'good');
+        getItemById(itemId, true);
       } else {
-        hasRequested.current.byUser[userId] = true;
-      }
-      let handler;
-      const base = {
-        apiVersion: 'v2',
-        clientName: 'Pattern',
-        env: PLAID_ENV,
-        product: ['transactions'],
-      };
-      const onLoad = () => {
-        logEvent('onLoad', {});
-        if (isUpdate) {
-          dispatch([types.LINK_UPDATE_MODE_LOADED, {id: itemId}]);
-        } else {
-          dispatch([types.LINK_LOADED, {id: userId}]);
-        }
-      };
-      const onSuccess = async (
-        publicToken,
-        {institution, accounts, link_session_id}
-      ) => {
-        logEvent('onSuccess', {institution, accounts, link_session_id});
-        await postLinkEvent({
-          userId,
-          link_session_id,
-          type: 'success',
-        });
-        if (isUpdate) {
-          await setItemState(itemId, 'good');
-          getItemById(itemId, true);
-        } else {
-          await exchangeToken(publicToken, institution, accounts, userId);
-          getItemsByUser(userId, true);
-        }
-
-        if (window.location.pathname === '/') {
-          window.location.href = `/user/${userId}/items`;
-        }
-      };
-      const onExit = async (
-        error,
-        {institution, link_session_id, request_id}
-      ) => {
-        logEvent('onExit', {
-          error,
-          institution,
-          link_session_id,
-          request_id,
-        });
-        const eventError = error || {};
-        await postLinkEvent({
-          userId,
-          link_session_id,
-          request_id,
-          type: 'exit',
-          ...eventError,
-        });
-        if (error != null && error.error_code === 'INVALID_LINK_TOKEN') {
-          // gets rid of the old iframe
-          handler.destroy();
-          await initializeLink();
-        }
-      };
-      const baseConfigs = {
-        ...base,
-        onEvent: logEvent,
-        onLoad: onLoad,
-        onSuccess: onSuccess,
-        onExit: onExit,
-      };
-
-      const initializeLink = async () => {
-        const linkTokenResponse = await getLinkToken({itemId, userId});
-        handler = await window.Plaid.create({
-          ...baseConfigs,
-          token: await linkTokenResponse.data.link_token,
-        });
+        await exchangeToken(publicToken, institution, accounts, userId);
+        getItemsByUser(userId, true);
       }
 
-      await  initializeLink();
-      if (isUpdate) {
-        dispatch([types.LINK_UPDATE_MODE_CREATED, {id: itemId, handler}]);
-      } else {
-        dispatch([types.LINK_CREATED, {id: userId, handler}]);
+      if (window.location.pathname === '/') {
+        window.location.href = `/user/${userId}/items`;
```

Comment:
this app uses react router, so this is not the right way to check the current route or change the current route. Can you look into using the built in react-router functionality instead such as the `useLocation` hook, the `useHistory`, and `history.push` hooks / functionality

Reply:
yes, this was existing Pattern code. I will look into using hooks